lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -146,6 +144,29 @@ def proj_transform(xs, ys, zs, M):
 transform = proj_transform
 
 
+def proj_transform_vectors(vecs, M):
+    """Vector version of ``project_transform`` able to handle MaskedArrays.
+
+    Parameters
+    ----------
+    vecs : 3xN np.ndarray or np.ma.MaskedArray
+        Input vectors
+
+    M : 4x4 np.ndarray
+        Projection matrix
+    """
+    is_masked = isinstance(vecs, np.ma.MaskedArray)
+    if is_masked:
+        mask = vecs.mask
+    vecs = np.concatenate([vecs, np.ones((1, *vecs.shape[1:]))], axis=0)
+    product = np.dot(M, vecs)
```

Comment:
Faster as:
```suggestion
    vecs_pad = np.empty((vecs.shape[1] + 1,) + vecs.shape[1:])
    vecs_pad[:-1] = vecs
    vecs_pad[-1] = 1
    product = np.dot(M, vecs_pad)
```

Reply:
It seems like your suggested patch has a small mistake in that it should be `vecs.shape[0] + 1`, so I've pushed a new commit instead of accepting this.